src/utilities.jl

```diff
@@ -38,9 +38,19 @@ function is_lower_snake_case_alphanumeric(x::AbstractString, also_allow=())
            all(i -> i in ALPHANUMERIC_SNAKE_CASE_CHARACTERS || i in also_allow, x)
 end
 
-####
-#### zstd_compress/zstd_decompress
-####
+#####
+##### arrrrr i'm a pirate
+#####
+
+const NamedTupleTimeSpan = NamedTuple{(:start, :stop),Tuple{Nanosecond,Nanosecond}}
+
+TimeSpans.istimespan(::NamedTupleTimeSpan) = true
+TimeSpans.start(x::NamedTupleTimeSpan) = x.start
+TimeSpans.stop(x::NamedTupleTimeSpan) = x.stop
+
+#####
+##### zstd_compress/zstd_decompress
+#####
 
 function zstd_compress(bytes::Vector{UInt8}, level=3)
     compressor = ZstdCompressor(; level=level)
```

Comment:
ah, probably not a bad idea. I wonder what the perf difference is between multithreading on the Julia side vs. zstd's built-in multithreading (not exposed via CodecZstd IIRC; for all i know it might be the same thing under the hood anyway)

Reply:
one possible advantage of doing it Julia side is composing with other threaded Julia functions, if it's nested or such. (Though I'm not totally sure how tuned that stuff is yet anyway)